From 63bae97cc0cc0fcd6a250f5b17bd0445b6fff472 Mon Sep 17 00:00:00 2001
From: =?utf-8?q?Marc-Andr=C3=A9=20Lureau?= <marc-andre.lureau@nokia.com>
Date: Tue, 17 Feb 2009 16:09:31 +0200
Subject: [PATCH 19/23] bluetooth: ignore a SCO socket read() bug...

This error happen frequently, it looks really suspicious. Could it be
a kernel bug? we will need a simpler test case.

(poll() return POLLIN on stream_fd, then read() == -1, but errno is
not set). Btw, maybe the read l == 0 case need to be handled
differently?
---
 src/modules/bluetooth/module-bluetooth-device.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/modules/bluetooth/module-bluetooth-device.c b/src/modules/bluetooth/module-bluetooth-device.c
index ee0651d..6378933 100644
--- a/src/modules/bluetooth/module-bluetooth-device.c
+++ b/src/modules/bluetooth/module-bluetooth-device.c
@@ -1077,7 +1077,7 @@ static void thread_func(void *userdata) {
             if (pollfd->revents & POLLIN) {
 
                 if (hsp_process_push(u) < 0)
-                    goto fail;
+                    /* goto fail */;
 
                 /* We just read something, so we are supposed to write something, too */
                 do_write = TRUE;
-- 
1.6.1.224.gb56c

